### DISCUSSION TITLE
Allow to use UUID for whole database as primary key.

### DISCUSSION THREAD

**Discussion by serhiicherepanov** [directus/directus#10919] [Feature Requests]
Hi, 

I looking for a way to remove all autoincrement fields for a database.

Because it creates a lot of issue in cases when we need to maintenance several independent instances and need to apply database migration as plain sql scripts. Using of uuid will resolve all conflicts in the directus Data structures and in the custom collections.

Currently we use uuids for collections, translations and relations. But it still not possible in system tables. Also the numeric id still required to correct working of agregation.

Thanks!

**Comment by rijkvanzanten** [comment]
What do you mean by 

> Also the numeric id still required to correct working of agregation.

?

**Reply by serhiicherepanov** [reply]
@rijkvanzanten As I know the graphql aggregation functions like `count` does not working for varchar fields. 
We faced with this issue previously and decide to leave numeric ids because directus potentially has dependencies on this.

**Reply by rijkvanzanten** [reply]
Count should work for everything, though you're right in saying you can't get certain functions, like Average, Min, or Max, with a non-numeric value

**Reply by serhiicherepanov** [reply]
Ok, I will check it in current release. But the several things still required the auto increment ids, for example translations or relaions tables. It can't be changed from the UI during creation and required schema uodate directly after creation. Would be fine to have ability to switch primary key behaviors to uuid for whole project.

**Reply by rijkvanzanten** [reply]
> for example translations or relaions tables. It can't be changed from the UI during creation

Only when you use the auto-created translations and junction tables 🙂 There's nothing stopping you from creating those collections manually first, and then using them in the relationship setup for the translations field.

(though yes I do realize it's easier if you could simply tell it to use UUIDs 👍🏻 )

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
